tests/ClientTest.php

```diff
@@ -975,17 +976,19 @@ public function sampleRateAbsoluteDataProvider()
 
     public function testSetAllObjectSerialize()
     {
-        $client = ClientBuilder::create()->getClient();
-
-        $client->setAllObjectSerialize(true);
-
-        $this->assertTrue($client->getSerializer()->getAllObjectSerialize());
-        $this->assertTrue($client->getReprSerializer()->getAllObjectSerialize());
+        $serializer1 = $this->prophesize(SerializerInterface::class);
```

Comment:
I didn't know we use `Prophecy`. As I never saw anything like this in the unit tests codebase can we change this to use the classic mock builder of PHPUnit? At least we keep everything consistent

Reply:
I find the mock builder less robust and reliable. I would prefer to migrate toward Prophecy, which is BTW included in PHPUnit anyway.